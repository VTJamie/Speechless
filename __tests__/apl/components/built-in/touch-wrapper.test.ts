import {
  APLText,
  APLTouchWrapper,
  DoublePressGesture,
  LongPressGesture,
  SendEventCommand,
  SwipeAwayGesture,
  TapGesture
} from '../../../../lib/apl';
import { matchSnapshot } from '../../../helper';
matchSnapshot(
  new APLTouchWrapper({
    items: [
      new APLText({
        text: 'placeholder',
        onCursorEnter: [
          new SendEventCommand({
            uniqueId: '1',
            handler: (input) => input.responseBuilder.getResponse()
          })
        ],
        onCursorExit: [
          new SendEventCommand({
            uniqueId: '1',
            handler: (input) => input.responseBuilder.getResponse()
          })
        ],
        onMount: [
          new SendEventCommand({
            uniqueId: '1',
            handler: (input) => input.responseBuilder.getResponse()
          })
        ]
      })
    ],
    gestures: [
      new TapGesture({
        onTap: [
          new SendEventCommand({
            uniqueId: '1',
            handler: (input) => input.responseBuilder.getResponse()
          })
        ]
      }),
      new DoublePressGesture({
        onDoublePress: [
          new SendEventCommand({
            uniqueId: '1',
            handler: (input) => input.responseBuilder.getResponse()
          })
        ],
        onSinglePress: [
          new SendEventCommand({
            uniqueId: '1',
            handler: (input) => input.responseBuilder.getResponse()
          })
        ]
      }),
      new LongPressGesture({
        onLongPressEnd: [
          new SendEventCommand({
            uniqueId: '1',
            handler: (input) => input.responseBuilder.getResponse()
          })
        ],
        onLongPressStart: [
          new SendEventCommand({
            uniqueId: '1',
            handler: (input) => input.responseBuilder.getResponse()
          })
        ]
      }),
      new SwipeAwayGesture({
        direction: 'left',
        onSwipeDone: [
          new SendEventCommand({
            uniqueId: '1',
            handler: (input) => input.responseBuilder.getResponse()
          })
        ],
        onSwipeMove: [
          new SendEventCommand({
            uniqueId: '1',
            handler: (input) => input.responseBuilder.getResponse()
          })
        ]
      })
    ]
  }),
  10
);
