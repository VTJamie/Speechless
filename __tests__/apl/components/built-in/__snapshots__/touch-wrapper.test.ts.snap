// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APLTouchWrapper: APLTouchWrapper 1`] = `
{
  "gestures": [
    {
      "onTap": [
        {
          "arguments": [
            "1",
          ],
          "type": "SendEvent",
        },
      ],
      "type": "Tap",
    },
    {
      "onDoublePress": [
        {
          "arguments": [
            "1",
          ],
          "type": "SendEvent",
        },
      ],
      "onSinglePress": [
        {
          "arguments": [
            "1",
          ],
          "type": "SendEvent",
        },
      ],
      "type": "DoublePress",
    },
    {
      "onLongPressEnd": [
        {
          "arguments": [
            "1",
          ],
          "type": "SendEvent",
        },
      ],
      "onLongPressStart": [
        {
          "arguments": [
            "1",
          ],
          "type": "SendEvent",
        },
      ],
      "type": "LongPress",
    },
    {
      "direction": "left",
      "onSwipeDone": [
        {
          "arguments": [
            "1",
          ],
          "type": "SendEvent",
        },
      ],
      "onSwipeMove": [
        {
          "arguments": [
            "1",
          ],
          "type": "SendEvent",
        },
      ],
      "type": "SwipeAway",
    },
  ],
  "items": [
    {
      "onCursorEnter": [
        {
          "arguments": [
            "1",
          ],
          "type": "SendEvent",
        },
      ],
      "onCursorExit": [
        {
          "arguments": [
            "1",
          ],
          "type": "SendEvent",
        },
      ],
      "onMount": [
        {
          "arguments": [
            "1",
          ],
          "type": "SendEvent",
        },
      ],
      "text": "placeholder",
      "type": "Text",
    },
  ],
  "type": "TouchWrapper",
}
`;
